pilot/pkg/networking/v1alpha3/listener.go

```diff
@@ -71,11 +73,9 @@ func BuildListeners(env model.Environment, node model.Proxy) ([]*xdsapi.Listener
 	switch node.Type {
 	case model.Sidecar:
 		return buildSidecarListeners(env, node)
-	case model.Router:
+	case model.Router, model.Ingress:
```

Comment:
I think the `model.Ingress` case should be removed. I thought `model.Ingress` was going to be replaced entirely by `model.Router`. Pilot should not need to know whether it is handling gateway+virtual service CRDs only or CRDs and k8s ingress resources that have been converted to gateway+virtual services.

Reply:
this is only so that the ingress controller can be recognized. if we update the ingress controller's node ID, then we are golden.